hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,256 @@
+
+Cold boot, also known as POR (power on reset) is the first reset state of the design.
+The power manager sequences the design from a freshly reset state to an active state where software can be initialized.
+
+- Low power entry is the process in which the device enters one of two low power modes (sleep or deep sleep).
+- Low power exit is the process in which the device exits low power mode and returns to active state.
+- Low power entry is always initiated by software, while low power exit is always initiated by a previously setup hardware event such as pins or internal timers.
+- The power manager processes the software and hardware requests to perform the appropriate actions.
+
+Reset scenarios refer to non-POR events that cause the device to reboot.
+There are various stimuli that can cause such a reset, ranging from external user input to watchdog timeout.
+The power manager processes the reset request and brings the device to an appropriate state.
+
+
+# Theory Of Operation
+
+The power manager performs the following functions:
+- Turn on/off power domain(s).
+- Control root resets with the reset manager.
+- Control root clock enables with AST and clock manager.
+- Sequence various power up activities such as OTP sensing, life cycle initiation and releasing software to execute.
+
+If there is a need to perform ROM hashing as an integrity check in the future, that will also be managed by the power manager.
+
+## Block Diagram
+
+See the below high level block diagram that illustrates the connections between the power manager and various system components.
+Blocks outlined with a solid magenta line are always on; while blocks outlined with a dashed magenta line are a mix of components that are and those that are not.
+
+![Power Manager Connectivity Diagram](pwrmgr_connectivity.svg)
+
+
+## Overall Sequencing
+
+The power manager contains two state machines.
+One operates on the always-on slow clock (this clock is always running and usually measured in KHz) and is responsible for turning faster clocks on and off and managing the power domains.
+The other operates on a normal fixed clock (usually measured in MHz) and is responsible for everything else in the power sequence.
+
+The following diagram breaks down the general functionality of both.
+The state machines are colored based on their clock domains.
+The green state machine is clocked by the normal fixed domain, while the organge state machine is clocked by the slow domain.
+
+![Power Manager FSMs](pwrmgr_fsms.svg)
+
+
+Note, most of the states are transitional states, and only the following state combinations are resting states.
+
+
+*   Slow FSM `Idle` and fast FSM `Active`
+*   Slow FSM `Low_Power` and fast FSM `Low_power`
+
+The slow FSM `Low_power` and fast FSM `Active` states specifically are concepts useful when examining [reset handling](#Reset-Request-Handling).
```

Comment:
Sorry for the nitpicks, but I don't think the capitalisation is consistent for `Low_power` / `Low_Power` here. Also, do you need the underscores? I think you've got spaces in the diagram.

Reply:
good catch, i didnt even notice some were capitalized and some were not.